compiler/rustc_passes/src/stability.rs

```diff
@@ -655,6 +653,44 @@ impl<'tcx> Visitor<'tcx> for MissingStabilityAnnotations<'tcx> {
     // stable (assuming they have not inherited instability from their parent).
 }
 
+struct CheckStableConstImplTrait<'tcx> {
+    tcx: TyCtxt<'tcx>,
+}
+
+impl<'tcx> ItemLikeVisitor<'tcx> for CheckStableConstImplTrait<'tcx> {
+    fn visit_item(&mut self, item: &'tcx Item<'tcx>) {
+        if !matches!(
+            item.kind,
+            hir::ItemKind::Impl(hir::Impl {
+                of_trait: Some(_),
+                constness: hir::Constness::Const,
+                ..
+            })
+        ) {
+            return;
+        }
+
+        if self.tcx.lookup_const_stability(item.def_id).map_or(false, |stab| stab.is_const_stable())
```

Comment:
I wasn't sure — is using `tcx.lookup_const_stability()` or `tcx.has_attr()` preferred? In this situation I don't believe there's a semantic difference.